src/js/auth.js

```diff
@@ -29,24 +37,53 @@ function getToken() {
     });
 }
 
+/*
+ * Modifies the current history state
+ */
+function replaceState(state) {
+  window.history.replaceState({}, document.title, state);
```

Comment:
we were doing this same `history.replaceState` in a few places, but this makes it clearer what it's doing in context